sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathProperties.cs

```diff
@@ -171,6 +171,41 @@ public class PathProperties
         /// </summary>
         public DateTimeOffset AccessTierChangedOn { get; internal set; }
 
+        /// <summary>
+        /// If this path represents a directory.
+        /// </summary>
+        public bool IsDirectory
+        {
+            get
+            {
+                Metadata.TryGetValue(Constants.DataLake.IsDirectoryKey, out string isDirectory);
+                if (isDirectory != null && isDirectory == bool.TrueString.ToLowerInvariant())
+                {
+                    return true;
+                }
+                return false;
+            }
+            internal set
+            {
+                // True
+                if (value)
+                {
+                    if (!Metadata.ContainsKey(Constants.DataLake.IsDirectoryKey))
+                    {
+                        Metadata.Add(Constants.DataLake.IsDirectoryKey, bool.TrueString.ToLowerInvariant());
```

Comment:
Why are we doing the check?  Why not just set `Metadata[key] =  bool.TrueString.ToLowerInvariant()`?

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.add?view=netcore-2.0

"ArgumentException - An element with the same key already exists in the Dictionary<TKey,TValue>."

Reply:
Sure but use the indexer instead of `Add`.  That just overwrites.  https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.item?view=netcore-2.0

Reply:
fixed.